### DISCUSSION TITLE
Hook when donations are added with renewal status

### DISCUSSION THREAD

**Discussion by 007amrit** [impress-org/givewp#7142] [Developer Q&A]
Hey 

is there are any hook like 

function your_function( $donation_id, $new_status, $old_status ) {
    your code here
}
add_action( 'give_update_payment_status', 'your_function', 200, 3 );

which we can use 

when renewal happens on stripe and it pushes donation in givewp with status "renewal"

**Comment by 007amrit** [comment]
hey thanks a lot for the prompt reply

Yes I am doing some custom code whenever a donation is added
it works well for all the donation create via donation form 
using hook give_update_payment_status

but I want to perform same custom code for each renewal donation as well

and it seems give_update_payment_status is not firing

**Comment by kjohnson** [comment]
The Donation Model has two hooks to consider:

- `givewp_donation_updating`
- `givewp_donation_updated`

Each hook is passed a single argument, which is the Donation Model.

Reference: https://github.com/impress-org/givewp/blob/develop/src/Donations/Repositories/DonationRepository.php#L242

**Reply by 007amrit** [reply]
@kjohnson 

givewp_donation_updating  hook didn't fire on renewal 
or perhaps did I use it in wrong way? can you pleae check answer below?

but give_insert_payment  did work :)

**Comment by 007amrit** [comment]
Thanks a lot @kjohnson 

So this 'givewp_donation_updated'  only fires when renewal is done?

does this not fire for normal donation? because I didn't fire. 

```
function my_custom_donation_updated_action( $donation_id ) {
       $to = 'myemail@gmail.com';
	$subject = 'Renewal test';
	$body = 'subscription test successful'; 
	wp_mail( $to, $subject, $body );	
}
add_action( 'givewp_donation_updated', 'my_custom_donation_updated_action', 10, 1 );

I hope it works for renewal .. will wait for renwal to happend tomorrow
```

**Comment by 007amrit** [comment]
But I was looking further 

I think this should work

```
function my_custom_new_donation_action( $payment_id, $payment_data ) {
    // Your custom code here

    // Example: Update some custom meta or perform an action
    update_post_meta( $payment_id, 'my_custom_meta_key', 'my_custom_value' );
}

add_action( 'give_insert_payment', 'my_custom_new_donation_action', 10, 2 );
```
